<!-- -*- mode: XML -*- -->
<launch>

    <arg name="map_frame" default="map"/>
    <arg name="odom_frame" default="odom"/>
    <arg name="base_frame" default="base_link"/>

    ################ car node ################
    <node pkg="robot" name="car" type="car" output="screen">
        <param name="map_frame" value="$(arg map_frame)"/>
        <param name="odom_frame" value="$(arg odom_frame)"/>
        <param name="base_frame" value="$(arg base_frame)"/>
        <!-- Initial pose of car -->
        <param name="x" value="8.0"/>
        <param name="y" value="8.0"/>
        <param name="theta" value="90.0"/>
        <!-- Initial velocity of car -->
        <param name="vx" value="0.5"/>
        <param name="vy" value="0"/>
        <param name="vth" value="0.5"/>
        <param name="duration" value="0.1"/>
        <!-- Polygon of car -->
        <rosparam param="polygon">[0.2, -0.2, 0.2, 0.2, -0.2, 0.2, -0.2, -0.2]</rosparam>
    </node>

    ################ keyboardcontrol node ################
    <node pkg="robot" name="keyboardcontrol" type="keyboardcontrol" output="screen">
        <rosparam param="KP">[1.0, 1.0, 1.0, 0.5]</rosparam>
        <param name="rate" value="10"/>
    </node>

    ################ Map server ################
    <node pkg="map_server" name="map_server" type="map_server" args="$(find robot)/map/house.yaml"/>

    ################ lidar node ################
    <node pkg="robot" name="lidar" type="lidar_node.py" output="screen">
        <remap from="odom" to="car/odom"/>
        <param name="map_frame" value="$(arg map_frame)"/>
        <param name="odom_frame" value="$(arg odom_frame)"/>
        <param name="base_frame" value="$(arg base_frame)"/>
        <param name="publish_frame" value="$(arg base_frame)"/>
        <param name="rate" value="10"/>
        <param name="occ_thresh" value="50"/>
        <param name="min_angle" value="0.0"/>
        <param name="max_angle" value="360.0"/>
        <param name="step_angle" value="1.0"/>
        <param name="min_range" value="0.0"/>
        <param name="max_range" value="5.0"/>     
        <param name="min_height" value="0"/>
        <param name="max_height" value="0.1"/>
        <param name="step_height" value="0.2"/>     
    </node>

    ################ rviz node ################
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find robot)/rviz/car.rviz"/>

</launch>
